### PR TITLE
Add awkward workaround for Chronic bug

### PR DESCRIPTION
**Pull request by JacobEvelyn**
Resolves #180

---

Hi there! Thanks so much for submitting a pull request!

Let's just make sure together that all of these boxes are checked before we
merge this change:

- [x] The code in these changes works correctly.
- [x] Code has tests as appropriate.
- [x] Code has been reviewed by @JacobEvelyn.
- [x] All tests pass on Travis CI.
- [x] Rubocop reports no issues on Travis CI.
- [x] The `README.md` file is updated as appropriate.

Don't worry—this list will get checked off in no time!

### CODE REVIEW

File: lib/friends/activity.rb [hunk 1]
```diff
@@ -34,9 +34,25 @@ def initialize(str: "")
       # Partition lets us parse "Today" and "Today: I awoke." identically.
       date_s, _, description = str.partition(DATE_PARTITION)
 
-      # rubocop:disable Lint/AssignmentInCondition
-      if time = (date_s =~ /^\d{4}-\d{2}-\d{2}$/ ? Time : Chronic).parse(date_s)
-        # rubocop:enable Lint/AssignmentInCondition
+      time = if date_s =~ /^\d{4}-\d{2}-\d{2}$/
```
Comment (greyhillman) [inline lib/friends/activity.rb]: I would make this a simple `if` block. Like:
```ruby
if date_s ...
    time = Time.parse(date_s)
else
    past_time = Chronic.parse(date_s, content: :past)
end
```

Then down below, have something like
```ruby
if past_time
  ...
else if time
  ...
else
 ...
end
```

To me, that reads better.

File: lib/friends/activity.rb [hunk 2]
```diff
@@ -34,9 +34,25 @@ def initialize(str: "")
...
+               past_time = Chronic.parse(date_s, context: :past)
+             end
+
+      if time
+        # Remove this when Chronic's `context: :past` actually works.
+        if time > Time.now
+          # I don't know of a better way of getting the same date but last year
+          # consistently (including leap years, etc.). Reparsing a new string
+          # is kind of ugly but it works.
+          correct_year = past_time.strftime("%Y").to_i - 1
```
Comment (greyhillman) [inline lib/friends/activity.rb]: I think this can be simplified to:
```ruby
time = Time.local(past_time.year - 1, past_time.month, past_time.day)
```

Comment (JacobEvelyn) [inline lib/friends/activity.rb]: Ah, yes! That's much nicer.

File: lib/friends/activity.rb (hunk 1 above)
Comment (JacobEvelyn) [inline lib/friends/activity.rb]: Agreed, good suggestion!

**Comment by JacobEvelyn** [comment]
Thanks so much for the code review! Just letting you know that it sounds like someone's rewriting Chronic and I'm waiting to hear back [here](https://github.com/mojombo/chronic/issues/361) on whether I can just release a gem containing the fix so we don't have to do this workaround. Unfortunately I'll be a bit hard to get ahold of over the next few days so this will probably not be fixed and released until next week sometime.

File: lib/friends/activity.rb (hunk 1 above)
Comment (JacobEvelyn) [inline lib/friends/activity.rb]: I actually ended up reworking this a little bit. Definitely agreed that the original was no good.

**Comment by JacobEvelyn** [comment]
This is now fixed in version 0.33. Thanks so so much for all the help, @greyhillman!

**Pull request merged**